app/controllers/reservations_controller.rb

```diff
@@ -21,20 +20,20 @@ def edit
 
   def create
     authorize Reservation
+
     @reservation = Reservation.new(screening_id: params[:screening_id], user_id: current_user.id, status: :created)
 
-    if !params.key?(:seats)
-      @reservation.errors.add(:base, 'You have to choose at least one seat')
-      render :new, status: :unprocessable_entity
-    else
-      @reservation.save
-      create_tickets
+    if CreateReservation.new(@reservation, params).call
       redirect_to movies_path, notice: 'Reservation successfully created'
+    else
+      @reservation.errors.add(:base, 'You have to choose at least one seat')
+      render :new, status: :unprocessable_entity and return
```

Comment:
What is the purpose of `and return` here?

```suggestion
      render :new, status: :unprocessable_entity
```

Reply:
I guess I misspelled, already deleted that